android/app/src/main/java/com/reactnativenavigation/modal/RnnModal.java

```diff
@@ -87,4 +105,17 @@ public void onBackPressed() {
     public void onDismiss(DialogInterface dialog) {
         ModalController.getInstance().remove();
     }
+
+    @Override
+    public void dismiss(){
+        super.dismiss();
+        if (mContext != null && !mContext.isFinishing()) {
+            mContext.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    updateToolbar(mOriginalScreen, mOriginalToolbar);
```

Comment:
If the Modal is dismissed, the previous toolbar already has the correct style and buttons. Do we call `updateToolbar` only for `mContext.setSupportActionBar(toolbar);` ?